File: sdkwork-ai-sdk-java/src/main/java/com/sdkwork/ai/api/ModerationApi.java
package com.sdkwork.ai.api;

import com.sdkwork.ai.http.HttpClient;
import com.sdkwork.ai.model.*;
import java.util.List;
import java.util.Map;

public class ModerationApi {
    private final HttpClient client;
    
    public ModerationApi(HttpClient client) {
        this.client = client;
    }

    /** Create moderation */
    public ModerationResponse createModeration(ModerationRequest body) throws Exception {
        return (ModerationResponse) client.post(ApiPaths.aiPath("/moderations"), body);
    }
}

File: sdkwork-ai-sdk-swift/Sources/API/ModerationApi.swift
import Foundation

public class ModerationApi {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Create moderation
    public func createModeration(body: ModerationRequest) async throws -> ModerationResponse? {
        let response = try await client.post(ApiPaths.aiPath("/moderations"), body: body)
        return response as? ModerationResponse
    }
}
